tests/track/loader_test.py

```diff
@@ -1357,6 +1357,84 @@ def test_filters_tasks(self):
 
+    def test_filters_exclude_tasks(self):
+        track_specification = {
+            "description": "description for unit test",
+            "indices": [{"name": "test-index", "auto-managed": False}],
+            "operations": [
+                {
+                    "name": "bulk-index",
+                    "operation-type": "bulk"
+                },
+                {
+                    "name": "node-stats",
+                    "operation-type": "node-stats"
+                },
+                {
+                    "name": "cluster-stats",
+                    "operation-type": "custom-operation-type"
+                },
+                {
+                    "name": "match-all",
+                    "operation-type": "search",
+                    "body": {
+                        "query": {
+                            "match_all": {}
+                        }
+                    }
+                },
+            ],
+            "challenges": [
+                {
+                    "name": "default-challenge",
+                    "schedule": [
+                        {
+                            "parallel": {
+                                "tasks": [
+                                    {
+                                        "name": "index-1",
+                                        "operation": "bulk-index",
+                                    },
+                                    {
+                                        "name": "index-2",
+                                        "operation": "bulk-index",
+                                    },
+                                    {
+                                        "name": "index-3",
+                                        "operation": "bulk-index",
+                                    },
+                                    {
+                                        "name": "match-all-parallel",
+                                        "operation": "match-all",
+                                    },
+                                ]
+                            }
+                        },
+                        {
+                            "operation": "node-stats"
+                        },
+                        {
+                            "name": "match-all-serial",
+                            "operation": "match-all"
+                        },
+                        {
+                            "operation": "cluster-stats"
+                        }
+                    ]
+                }
+            ]
+        }
+        reader = loader.TrackSpecificationReader()
+        full_track = reader("unittest", track_specification, "/mappings")
+        self.assertEqual(4, len(full_track.challenges[0].schedule))
+
+        filtered = loader.filter_tasks(full_track, [track.TaskNameFilter("index-3")], exclude=True)
+
+        schedule = filtered.challenges[0].schedule
```

Comment:
If we exclude one of the parallel tasks then nothing in that group runs.. should I correct the behaviour for that or does it make sense like that?

Reply:
I think if we exclude a single task in a parallel element, only that single one should be excluded but all others should still be included.

Reply:
Fixed.

Reply:
I had another look and I meant something else than what is implemented at the moment. I have visualized the challenge `index-and-query-logs-fixed-daily-volume` from the eventdata track here which contains such a parallel element (I've simplified it to only simulate a single day):

```
1. measure-maximum-utilization (8 clients)
2. delete-measurement-index
3. 5 parallel tasks (12 clients):
	3.1 bulk-index-logs-100%-utilization (8 clients)
	3.2 current-kibana-traffic-country-dashboard_60m-querying-100%-utilization
	3.3 current-kibana-discover_30m-querying-100%-utilization
	3.4 current-kibana-traffic-dashboard_30m-querying-100%-utilization
	3.5 current-kibana-content_issues-dashboard_30m-querying-100%-utilization
```

If we specify `--exclude-tasks="type:search"`, I expect the following schedule:

```
1. measure-maximum-utilization (8 clients)
2. delete-measurement-index
3. 1 parallel task (8 clients):
	3.1 bulk-index-logs-100%-utilization (8 clients)
```

but instead we get:

```
1. measure-maximum-utilization (8 clients)
2. delete-measurement-index
3. 5 parallel tasks (12 clients):
	3.1 bulk-index-logs-100%-utilization (8 clients)
	3.2 current-kibana-traffic-country-dashboard_60m-querying-100%-utilization
	3.3 current-kibana-discover_30m-querying-100%-utilization
	3.4 current-kibana-traffic-dashboard_30m-querying-100%-utilization
	3.5 current-kibana-content_issues-dashboard_30m-querying-100%-utilization
```

Can you please have another look?

Reply:
fyi, I've created this output with a new `info` subcommand that I've added to Rally. It already includes support for your change but as it is not merged yet I've only pushed it as draft PR https://github.com/elastic/rally/pull/850.

Reply:
Scratch that. As discussed offline, the operation type is `kibana` not `search`. So the filter is actually working fine. Sorry for the confusion. :)